File: haskell/fizzy/fizzy.hs
#!/usr/bin/env runhaskell

-- FizzBuzz with parallel processing
-- Andrew Pennebaker

import Control.Parallel.Strategies
import Control.Parallel

fizzBuzz :: [String]
fizzBuzz = (parMap rseq) fizzBuzz' [1 .. 100]
  where
    fizzBuzz' :: Int -> String
    fizzBuzz' x = case (x `mod` 3, x `mod` 5) of
      (0, 0) -> "FizzBuzz"
      (0, _) -> "Fizz"
      (_, 0) -> "Buzz"
      _ -> show x

main :: IO ()
main = putStrLn . unlines $ fizzBuzz

File: scripting/general/ruby/fizzbuzz/fizzy.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'parallel'

#
# FizzBuzz
#
class Fixnum
    def fizz
        x, y = self % 3, self % 5

        if [x, y] == [0, 0]
            'FizzBuzz'
        elsif x == 0
            'Fizz'
        elsif y == 0
            'Buzz'
        else
            to_s
        end
    end
end

def main
    Parallel.map(1.upto(100)) { |i| i.fizz }.each { |s| puts s }
end

main if $PROGRAM_NAME == __FILE__
